Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java

```diff
@@ -672,24 +672,32 @@ public void getCurrentPosition(String callback) {
     public void isMockLocation(String callback) {
         Activity activity = bridgeComponents.getActivity();
         if (!isLocationPermissionEnabled()) return;
-        if (client != null && activity != null)
+        if (client != null && activity != null) {
             client.getLastLocation()
                     .addOnSuccessListener(activity, location -> {
-                        boolean isMock;
+                        boolean isMock = false;
                         if (location != null) {
                             if (Build.VERSION.SDK_INT <= 30) {
                                 isMock = location.isFromMockProvider();
                             } else {
                                 isMock = location.isMock();
                             }
-                            if (callback != null) {
-                                String js = String.format(Locale.ENGLISH, "window.callUICallback('%s','%s');",
-                                        callback, isMock);
+                        }
+                        if (callback != null) {
+                            String js = String.format(Locale.ENGLISH, "window.callUICallback('%s','%s');",
+                                  callback, ((location != null) ? isMock : "failed"));
```

Comment:
failed can be false, don't make it unsafe as isMock is a boolean and in else condition we are sending a string. Failed can be treated as false